dotnet/WPF/VisualLayerIntegration/VisualLayerIntegration/BarGraphUtility/BarGraph.cs

```diff
@@ -244,108 +266,108 @@ public void DrawText(WindowRenderTarget renderTarget, string titleText, string x
+            renderTarget.Transform = SharpDX.Matrix3x2.Rotation((float)(-Math.PI / 2), new SharpDX.Vector2(0, containerHeight));
 
             renderTarget.BeginDraw();
 
-            renderTarget.DrawText(yAxisText, textFormatVertical, ClientRectangleYAxis, textSceneColorBrush);
+            renderTarget.DrawText(yAxisText, _textFormatVertical, ClientRectangleYAxis, _textSceneColorBrush);
 
             renderTarget.EndDraw();
 
             // Rotate the RenderTarget back.
-            renderTarget.Transform = Matrix3x2.Identity;
+            renderTarget.Transform = SharpDX.Matrix3x2.Identity;
         }
 
         //Dispose of resources.
         public void Dispose()
         {
-            textSceneColorBrush.Dispose();
-            textFormatTitle.Dispose();
-            textFormatHorizontal.Dispose();
-            textFormatVertical.Dispose();
+            _textSceneColorBrush.Dispose();
+            _textFormatTitle.Dispose();
+            _textFormatHorizontal.Dispose();
+            _textFormatVertical.Dispose();
         }
 
-        private Bar[] CreateBars(float[] data)
+        private Bar[] CreateBars(double[] data)
         {
-            //Clear hashmap.
-            barValueMap.Clear();
+            //Clear
+            _bars = new Bar[data.Length];
 
-            var barBrushHelper = new BarGraphUtility.BarBrushHelper(compositor);
+            var barBrushHelper = new BarGraphUtility.BarBrushHelper(_compositor);
             var brushes = new CompositionBrush[data.Length];
-            switch (graphBarStyle)
+            CompositionBrush brush = null;
+
+            switch (_graphBarStyle)
             {
                 case GraphBarStyle.Single:
-                    brushes = barBrushHelper.GenerateSingleColorBrush(data.Length, graphBarColors[0]);
+                    brush = barBrushHelper.GenerateSingleColorBrush(_graphBarColors[0]);
                     break;
                 case GraphBarStyle.Random:
                     brushes = barBrushHelper.GenerateRandomColorBrushes(data.Length);
                     break;
                 case GraphBarStyle.PerBarLinearGradient:
-                    brushes = barBrushHelper.GeneratePerBarLinearGradient(data.Length, graphBarColors);
+                    brush = barBrushHelper.GenerateLinearGradient(_graphBarColors);
                     break;
                 case GraphBarStyle.AmbientAnimatingPerBarLinearGradient:
-                    brushes = barBrushHelper.GenerateAmbientAnimatingPerBarLinearGradient(data.Length, graphBarColors);
+                    brush = barBrushHelper.GenerateAmbientAnimatingLinearGradient(_graphBarColors);
                     break;
                 default:
-                    brushes = barBrushHelper.GenerateSingleColorBrush(data.Length, graphBarColors[0]);
+                    brush = barBrushHelper.GenerateSingleColorBrush(_graphBarColors[0]);
```

Comment:
This is the same code as the GraphBarStyle.Single code. If that's intended then move the default label to the .Single label rather than repeating the code.

Reply:
updated